airbyte-webapp/src/components/ui/TagInput/TagInput.tsx

```diff
@@ -1,14 +1,19 @@
+import classNames from "classnames";
 import uniqueId from "lodash/uniqueId";
-import { KeyboardEventHandler, useMemo, useState } from "react";
-import { ActionMeta, GroupBase, MultiValue, OnChangeValue, StylesConfig } from "react-select";
+import { ComponentType, KeyboardEventHandler, useMemo, useState } from "react";
+import { ActionMeta, GroupBase, MultiValue, OnChangeValue, StylesConfig, components, InputProps } from "react-select";
 import CreatableSelect from "react-select/creatable";
 
 import styles from "./TagInput.module.scss";
 
-const components = {
+const overwrittenComponents = {
   DropdownIndicator: null,
 };
 
+const NumberInput: ComponentType<InputProps<Tag, true, GroupBase<Tag>>> = (
+  props: InputProps<Tag, true, GroupBase<Tag>>
+) => <components.Input {...props} type="number" className={classNames(props.className, styles.hideArrowButtons)} />;
```

Comment:
I found an issue when testing this locally: if you type an `e` into the input, it accepts it (I believe this is expected because `e` can be used to specify a number in scientific notation), but the character is halfway cut off in the input. I can type more numbers after the `e`, but they are still cut off, and I cannot erase them by pressing backspace. And if I try to hit enter, it shows me the default `Please enter a number` warning. 

Here is a loom to demonstrate: https://www.loom.com/share/9a0429912ed049968072cc3b760f9dfc

I think allowing `e` here is probably fine since that can technically still be used for a number, but the visual issue should probably try to be fixed? If it ends up being really difficult to fix then we can always do that in a separate PR

Reply:
Good call, this happens because a number input will set its `value` to `''` if it can't be parsed as a number which breaks react-select because it checks the `value` to layout it.

I changed the approach completely by parsing the input as a number before it gets put into the state and filtering out invalid entries. It kind of works, but it's not perfect either because it could be confusing if the user enters something, then hits Enter and it's suddenly gone again because it couldn't be parsed. On the plus side this approach still keeps `,`/`;` separated lists of numbers working.

It's a really small thing, but super annoying to solve. What do you think? Pretty much out of better ideas... The only other solution would be to drop this feature completely and only allow string inputs which then have to be parsed connector side (we can still do client side validation using a pattern).

Reply:
I think this solution is good, I like that is brings back the `,`/`;` functionality so it is more consistent with other types of arrays. 

I agree it is a little bit confusing that if you type in a string without a number and hit enter, it just disappears. But this is still a huge improvement on the page crashing and contains no weird visual bugs, so I think we should go with it.

To make this less confusing in the future, maybe as part of the field format changes in the Simplify connector form project, we should also show an indication of which type of values should go into these array inputs to make it clear to users that only numbers can be entered in this case.

Reply:
> To make this less confusing in the future, maybe as part of the field format changes in the Simplify connector form project, we should also show an indication of which type of values should go into these array inputs to make it clear to users that only numbers can be entered in this case.

I like that a lot, great idea.